### PR TITLE
ARROW-1875: [Java] Write 64-bit ints as strings in integration test JSON files

### PR DESCRIPTION
**Pull request by tianchen92**
Related to [ARROW-1875](https://issues.apache.org/jira/browse/ARROW-1875).
This is Java side implementation.

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java [hunk 1]
```diff
@@ -285,7 +285,7 @@ private void writeValueToGenerator(
           generator.writeNumber(UInt4Vector.getNoOverflow(buffer, index));
           break;
         case UINT8:
-          generator.writeNumber(UInt8Vector.getNoOverflow(buffer, index));
+          generator.writeString(UInt8Vector.getNoOverflow(buffer, index).toString());
```
Comment (emkornfield) [inline java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java]: I think this should stay as number

Comment (emkornfield) [reply java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java]: NM, naming conventions in java confused me.

**Comment by wesm** [comment]
I'm surprised that the integration tests pass. Is RapidJSON loose about parsing strings to numbers?

**Comment by tianchen92** [comment]
Also surprised me, I thought it would only pass if all the implementations work is done. Since it not break travis, could we just merge this into master or hold it in integration branch?

**Comment by kou** [comment]
Why does this pull request target to https://github.com/apache/arrow/tree/integration_ints_as_strings branch instead of master?
Should we use the branch? Can we remove the branch?

**Comment by emkornfield** [comment]
@kou we thought this change would need coordination between multiple languages.  It appears it might not. @tianchen92 do you want to try to make the pull request against master and see if the tests still pass?

**Comment by tianchen92** [comment]
> @kou we thought this change would need coordination between multiple languages. It appears it might not. @tianchen92 do you want to try to make the pull request against master and see if the tests still pass?

I think i will work fine in master, I just opened a new PR https://github.com/apache/arrow/pull/5002, close this one and will reopen if needed.

**Comment by kou** [comment]
@emkornfield Thanks for the answer. There is no problem if we do this intentionally. I thought that this is caused by error. Because we don't use a branch like this.
Sorry for missing a discussion for this branch usage.

**Pull request closed without merging**